src/bms/command/channel.rs

```diff
@@ -175,6 +173,87 @@ pub enum PlayerSide {
     Player2,
 }
 
+/// A channel ID used in BMS files to identify channel types.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct ChannelId([u8; 2]);
+
+impl std::fmt::Debug for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("ChannelId")
+            .field(&format!("{}{}", self.0[0] as char, self.0[1] as char))
+            .finish()
+    }
+}
+
+impl std::fmt::Display for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}", self.0[0] as char, self.0[1] as char)
+    }
+}
+
+impl TryFrom<[char; 2]> for ChannelId {
+    type Error = [char; 2];
+    fn try_from(value: [char; 2]) -> core::result::Result<Self, Self::Error> {
+        Ok(Self([
+            char_to_base62(value[0]).ok_or(value)?,
+            char_to_base62(value[1]).ok_or(value)?,
+        ]))
+    }
+}
+
+impl TryFrom<[u8; 2]> for ChannelId {
+    type Error = [u8; 2];
+    fn try_from(value: [u8; 2]) -> core::result::Result<Self, Self::Error> {
+        <Self as TryFrom<[char; 2]>>::try_from([value[0] as char, value[1] as char])
+            .map_err(|_| value)
+    }
+}
+
+impl<'a> TryFrom<&'a str> for ChannelId {
+    type Error = &'a str;
+    fn try_from(value: &'a str) -> core::result::Result<Self, Self::Error> {
+        if value.len() != 2 {
+            return Err(value);
+        }
+        let mut chars = value.bytes();
+        let [Some(ch1), Some(ch2), None] = [chars.next(), chars.next(), chars.next()] else {
+            return Err(value);
+        };
+        Self::try_from([ch1, ch2]).map_err(|_| value)
+    }
+}
```

Comment:
Instead of impl `TryFrom<&str>`, you should impl [`FromStr` trait](https://doc.rust-lang.org/std/str/trait.FromStr.html). It has a better type inference.